state/protocol/badger/mutator.go

```diff
@@ -477,7 +515,12 @@ func (m *FollowerState) insert(ctx context.Context, candidate *flow.Block, last
 
 	// trigger BlockProcessable for parent blocks above root height
 	if parent.Height > m.rootHeight {
-		m.consumer.BlockProcessable(parent)
+		m.consumer.BlockProcessable(parent, qc)
+	}
+
+	if certifyingQC != nil {
+		// trigger BlockProcessable for candidate block if it's certified
+		m.consumer.BlockProcessable(candidate.Header, certifyingQC)
```

Comment:
For followers, I think this will result in `BlockProcessable` being invoked (at least) twice for every incorporated block. Technically we expect consumers to handle duplicate events (for example in case of a restart), but I think it's worth the effort to avoid doing this for every block. 

One thing we could do is enqueue the `BlockProcessable` event while inserting the QC (only if the QC was inserted).